tests/ticket/search.rb

```diff
@@ -108,6 +285,399 @@
+        (result['message']).should.equal('INVALID_OWNER_FILTER')
+
+        result = request('/ticket/search', {
+            csrf_userid: $csrf_userid,
+            csrf_token: $csrf_token,
+            page: 1,
+            owners: "[30, 25]"
+        })
+        (result['status']).should.equal('fail')
+        (result['message']).should.equal('INVALID_OWNER_FILTER')
+
+        result = request('/ticket/search', {
+            csrf_userid: $csrf_userid,
+            csrf_token: $csrf_token,
+            page: 1,
+            owners: "ownerList"
+        })
+        (result['status']).should.equal('fail')
+        (result['message']).should.equal('INVALID_OWNER_FILTER')
+
+        result = request('/ticket/search', {
+            csrf_userid: $csrf_userid,
+            csrf_token: $csrf_token,
+            page: 1,
+            owners: -1
+        })
+        (result['status']).should.equal('fail')
+        (result['message']).should.equal('INVALID_OWNER_FILTER')
+
+        result = request('/ticket/search', {
+            csrf_userid: $csrf_userid,
+            csrf_token: $csrf_token,
+            page: 1,
+            owners: "{1, 3}"
+        })
+        (result['status']).should.equal('fail')
+        (result['message']).should.equal('INVALID_OWNER_FILTER')
+    end
+
+    it 'should success if the owners are valid' do
+        ownerId1 = 8
+        ownerId2 = 1
+        ownerId3 = 2
+
+        ownerIdList1 = [ownerId1].to_json
+        ownerIdList2 = [ownerId2, ownerId3, ownerId1].to_json
+        ownerIdList3 = [ownerId3, ownerId1].to_json
+
+        ownerId1 = ownerId1.to_s
+        ownerId2 = ownerId2.to_s
+        ownerId3 = ownerId3.to_s
+
+        for page in $pages
+            result = request('/ticket/search', {
+                csrf_userid: $csrf_userid,
+                csrf_token: $csrf_token,
+                page: page,
+                owners: ownerIdList1
+            })
+            (result['status']).should.equal('success')
+
+            tickets = result['data']['tickets']
+
+            for ticket in tickets
+                (ticket['owner']['id']).should.equal(ownerId1)
+            end
+
+            result = request('/ticket/search', {
+                csrf_userid: $csrf_userid,
+                csrf_token: $csrf_token,
+                page: page,
+                owners: ownerIdList2
+            })
+            (result['status']).should.equal('success')
+
+            tickets = result['data']['tickets']
+
+            for ticket in tickets
+                ticketOwnerId = ticket['owner']['id']
+                if ticketOwnerId === ownerId1
+                    ticketOwnerId.should.equal(ownerId1)
+                elsif ticketOwnerId === ownerId2
+                    ticketOwnerId.should.equal(ownerId2)
+                elsif ticketOwnerId === ownerId3
+                    ticketOwnerId.should.equal(ownerId3)
+                else
+                    1.should.equal(0)
+                end
```

Comment:
You could check if the expression `ticketOwnerId in ownerIdList2` (it's pseudo syntax, I don't know how to write that in Ruby) is true instead of the `1.should.equal(0)`. This way the error message would be better (than "error, 1 should be equal to 0"), and the code would end up being more concise.